pkgs/games/hmcl-bin/default.nix

```
@@ -0,0 +1,94 @@
+{ lib
+, stdenv
+, file
+, glib
+, fetchurl
+, makeWrapper
+, wrapGAppsHook
+, jdk
+, xorg
+, wayland
+, libpulseaudio
+, libglvnd
+, libGL
+, dconf
+, glfw
+, makeDesktopItem
+, copyDesktopItems
+, openal
+, alsa-lib
+}:
+
+stdenv.mkDerivation rec {
+  name = "hmcl-bin";
+  version = "3.5.3.223";
+
+  src = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/v${version}/HMCL-${version}.jar";
+    sha256 = "sha256-8g2FMvAiAKfxJUY0G7wl6d44wOpVsknFHGZ85IkOzFc=";
+  };
+
+  dontUnpack = true;
+
+  icon = fetchurl {
+    url = "https://aur.archlinux.org/cgit/aur.git/plain/craft_table.png?h=hmcl-bin";
+    sha256 = "sha256-KYmhtTAbjHua/a5Wlsak5SRq+i1PHz09rVwZLwNqm0w";
+  };
+
+  buildInputs = [ glib ];
+  nativeBuildInputs = [ jdk wrapGAppsHook makeWrapper file copyDesktopItems ];
```

```suggestion
  nativeBuildInputs = [ wrapGAppsHook makeWrapper copyDesktopItems ];
```
packages used in phases directly do not work with splicing and file is unused. Please also remove it from inputs.

---

Sorry, but the build failed after I removed jdk.

---

Then that was to much. I was just guessing.